### PR TITLE
New json files for MiniAODv4, minor updates in runner and README

### PR DESCRIPTION
**Pull request by uttiyasarkar**
1. New json files for 2023 are added for QCD,QCDmu,JetMET, and btag
2. 2022 jsons with MiniAODv4 samples updated
3. runner.py script is updated to pick up either conda/micromamba environment 
4. README updated stating that al9 is not recommended

Missing jsons:
mc_summer22EE_MINIAODv4_qcdmu
mc_summer22_MINIAODv4_qcdincl
mc_summer22EE_MINIAODv4_BTV

### CODE REVIEW

File: DY_test.txt
Comment (Ming-Yan) [inline DY_test.txt:1]: what is this file for?

File: metadata/test_data.json
Comment (Ming-Yan) [inline metadata/test_data.json:1]: what is this test data for ?

File: src/BTVNanoCommissioning/utils/compile_jec_test.py
Comment (Ming-Yan) [inline src/BTVNanoCommissioning/utils/compile_jec_test.py:1]: Is there any reason with this compile_jec_test?
I also see the flake8 error in the code checks

**Review by Ming-Yan — CHANGES_REQUESTED** [review]
Hi @uttiyasarkar thank you for the PR, looks good to me

I have some question regarding to the file names
1. why the file name is with v3?
2. if you want to modify the new file name instead of replacing the original file list, please also change recommended file name in readme.

File: DY_test.txt
Comment (uttiyasarkar) [reply DY_test.txt:1]: Duplicate files for test, can be removed

File: metadata/test_data.json
Comment (uttiyasarkar) [reply metadata/test_data.json:1]: Duplicate files for test, can be removed

File: src/BTVNanoCommissioning/utils/compile_jec_test.py
Comment (uttiyasarkar) [reply src/BTVNanoCommissioning/utils/compile_jec_test.py:1]: Also not required for the framework, I will remove it

File: metadata/test_data.json
Comment (uttiyasarkar) [reply metadata/test_data.json:1]: There are also other test_*.json duplicate files from the past. Do I clean them up as well or are they for some reason?
test_bta_run3.json
test_data.json
test_w_dj_ee.json
test_w_dj_e.json
test_w_dj_emu.json
test_w_dj_mu.json
test_w_dj_mumu.json

Comment (Ming-Yan) [reply metadata/test_data.json:1]: these are use to construct the ci pipeline, please leave it there

**Review by Ming-Yan — DISMISSED** [review]

**Comment by uttiyasarkar** [comment]
In the json filesets, we still have the following datasets missing:

1. data_2022_btagmu.json
2. data_2022_jetmet.json
3. data_2023_Mu.json
4. data_2023_em.json

**Comment by uttiyasarkar** [comment]
All jsons are prepared now. @Ming-Yan please review and it should be ready to be merged.

File: .sites_map.json
Comment (Ming-Yan) [inline .sites_map.json:1]: this seems reverse to your local one. might need the version I send to you
for example `"T2_BE_IIHE": {"/+(.*)": "root://maite.iihe.ac.be:1095//$1"}`

File: README.md [hunk 1]
```diff
@@ -241,10 +241,10 @@ However, some sites have certain restrictions for various reasons, in particular
 Memory usage is also useful to adapt to cluster. Check the memory by calling  `memory_usage_psutil()` from `helpers.func.memory_usage_psutil` to optimize job size. Example with `ectag_Wc_sf` summarized below.
  Type        |Array+Hist |  Hist only| Array Only|
 | :---:   | :---: | :---: | :---: |
-DoubleMuon (BTA,BTV_Comm_v2)| 1243MB |	848MB	|1249MB|
+DoubleMuon (BTA,BTV_Comm_v3)| 1243MB |	848MB	|1249MB|
 DoubleMuon (PFCands, BTV_Comm_v1)|1650MB	|1274MB	|1632MB|
 DoubleMuon (Nano_v11)|1183MB|	630MB	|1180MB|
-WJets_inc (BTA,BTV_Comm_v2)| 1243MB	|848MB	|1249MB|
+WJets_inc (BTA,BTV_Comm_v3)| 1243MB	|848MB	|1249MB|
```
Comment (Ming-Yan) [inline README.md:247]: test are based on v2

File: README.md [hunk 2]
```diff
@@ -208,7 +208,7 @@ python runner.py --workflow valid --json metadata/$json file
 
 Based on Congqiao's [development](notebooks/BTA_array_producer.ipynb) to produce BTA ntuples based on PFNano.
 
-:exclamation: Only the newest version [BTV_Run3_2022_Comm_v2](https://github.com/cms-jet/PFNano/tree/13_0_7_from124MiniAOD) ntuples work. Example files are given in [this](metadata/test_bta_run3.json) json. Optimize the chunksize(`--chunk`) in terms of the memory usage. This depends on sample, if the sample has huge jet collection/b-c hardons. The more info you store, the more memory you need. I would suggest to test with `iterative` to estimate the size.
+:exclamation: Only the newest version [BTV_Run3_2022_Comm_v3](https://github.com/cms-jet/PFNano/tree/13_0_7_from124MiniAOD) ntuples work. Example files are given in [this](metadata/test_bta_run3.json) json. Optimize the chunksize(`--chunk`) in terms of the memory usage. This depends on sample, if the sample has huge jet collection/b-c hardons. The more info you store, the more memory you need. I would suggest to test with `iterative` to estimate the size.
```
Comment (Ming-Yan) [inline README.md:211]: The newest version should not refers to PFNano but the btvnano development. https://github.com/cms-sw/cmssw/pull/43485

File: metadata/MC_Summer23_Run3BP_2023_qcd_BTV_Comm_v1_NanoV12.json
Comment (Ming-Yan) [inline metadata/MC_Summer23_Run3BP_2023_qcd_BTV_Comm_v1_NanoV12.json:1]: is there any recent to put as BP?

BTW, given Summer23 already refers to run3, I think Run3 is no longer needed in the syntax.

File: metadata/MC_Summer23_Run3BP_2023_qcdmu_BTV_Comm_v1_NanoV12.json
Comment (Ming-Yan) [inline metadata/MC_Summer23_Run3BP_2023_qcdmu_BTV_Comm_v1_NanoV12.json:1]: same comments with all BPix samples

**Review by Ming-Yan — CHANGES_REQUESTED** [review]
Hi @uttiyasarkar  thanks for the changes! 
Only one minor thing on naming was not clear, I think for the file name we would like to format as following.
`$CAMPAIGN` refers to the `--campaign` in `runner.py`
`$YEAR` refers to  `--year` in `runner.py`
`$SAMPLE` are the collections we collected for QCD/QCD-mu, rest of MC and dataset name
`$TAG_EXTENSTION` refers to the production string in our customize production. i.e. `BTV_Run3_2022_Comm_MINIAODv4` for our current [production](https://github.com/cms-btv-pog/btvnano-prod/blob/NanoAODv12_22Sep2023/crab_ymls/data_2022_MINIAODv4_MuonEG.yml#L12C18-L12C46)
```
MC: MC_$CAMPAIGN_$YEAR_$SAMPLE_$TAG_EXTENSION
data: data_$CAMPAIGN_$YEAR_$DATASET_$TAG_EXTENSION
```

File: metadata/MC_Summer23_Run3BP_2023_qcdmu_BTV_Comm_v1_NanoV12.json
Comment (Ming-Yan) [reply metadata/MC_Summer23_Run3BP_2023_qcdmu_BTV_Comm_v1_NanoV12.json:1]: Summer23_BPix, Summer23

File: metadata/MC_Summer23_Run3BP_2023_qcd_BTV_Comm_v1_NanoV12.json
Comment (uttiyasarkar) [reply metadata/MC_Summer23_Run3BP_2023_qcd_BTV_Comm_v1_NanoV12.json:1]: Fixed :)

File: .sites_map.json
Comment (uttiyasarkar) [reply .sites_map.json:1]: Reverted to the correct site list

File: README.md (hunk 2 above)
Comment (uttiyasarkar) [reply README.md:211]: Pointed to the recent framework https://github.com/cms-btv-pog/btvnano-prod

**Review by Ming-Yan — APPROVED** [review]
Thanks @uttiyasarkar looks good to me! we can merge this PR

**Pull request merged**